config/v1/types_project.go

```diff
@@ -20,8 +19,41 @@ type Project struct {
 }
 
 type ProjectSpec struct {
-	// project request message
-	// project request template
+	// DefaultNodeSelector holds default project node label selector
+	DefaultNodeSelector string `json:"defaultNodeSelector"`
+
+	// ProjectRequestMessage is the string presented to a user if they are unable to request a project via the projectrequest api endpoint
+	ProjectRequestMessage string `json:"projectRequestMessage"`
+
+	// ProjectRequestTemplate is the template to use for creating projects in response to projectrequest.
+	// It is in the format namespace/template and it is optional.
+	// If it is not specified, a default template is used.
+	ProjectRequestTemplate string `json:"projectRequestTemplate"`
```

Comment:
If this is a reference to an existing object, why not use `ObjectReference` consistently with other API?

Reply:
> If this is a reference to an existing object, why not use `ObjectReference` consistently with other API?

Because it is a specific kind of reference.  It's time to be opinionated.  This should point to a template in `openshift-config` namespace.

Reply:
I like my `ConfigMapNameReference` struct :smile:

Reply:
agree, I will update the comment